### ISSUE TITLE
UX: Accessibility of "Subscribe to page" and menus consistency

### ISSUE THREAD

**Issue by gawa971** [skyjake/lagrange#209]
Hello,

I struggled a lot to find the "Subscribe to page" feature. I think it should be easier to find.
* In the main hamburger menu, there are options to list bookmarks and show feed entries, but not to "Add bookmark" and "Subscribe to page"
* So I went looking for the bindings in the options. We can bind a shortcut to "Add bookmark" (default `Ctrl+D`), but not to "Subscribe to page"
* I took minutes to understand that the right click on the page offers a different menu with the options to "Bookmark page" and "Subscribe to page"
* Right now, right clicking on the page is the only way to learn about the shortcut for Subscribing (it shows the shortcut `Ctrl+Shift+D`)

**First enhancement**: Add the possibility to change the binding for "Subscribe feed" in the options. It would help users to learn about the shortcut at least.

Now I think I understand the rationale behind this: the right click is supposed to offer options related to the page and quick navigation while the hamburger menu offers more generic and global options.

That said, the rule is inconsistent because the hamburger menu has two options related to the page: "Copy page source" and "Save to download". Are those considered important enough as to appear in both menus ? More important than adding bookmark or subscribing ?

I think the distinction in the menus is not intuitive for new users. I discovered the very cool "Set Auto-Reload" feature and the "Import links as bookmarks" in the right-click menu by luck, by looking for the Subscribe feed.

**Second enhancement** (a proposal, just my opinion):
* It would make more sense to have all the options in the hamburger menu. That is the expected behaviour of all GUI software IMO. In most GUI softwares (firefox comes to mind), we go to this menu for the options we don't remember the shortcuts to,because we don't use these features often.
* There's no harm in features being duplicated in multiple menus. We could still keep the right-click menu. Many software do this.
* There are not many features in Lagrange so it would not clutter the hamburger menu.
* The only things that do not need to appear in the hamburger menu are the quick navigation features. Going to back/forward/parent/root are nice to have but you can do it by editing the url bar. They are therefore convenience, not unique feature. This is not the case with the other options (Set Auto-Reload, Subscribe, Import links as bookmarkl) where you "need" them to appear in the menu to use them.

TL;DR:
* Add the possibility to change the binding for "Subscribe feed" in the options
* Add in the hamburger menu important options, at least the "Add bookmark" and "Subscribe feed" options.

Thanks a lot for this awesome gemini browser by the way :pray:

**Comment by skyjake** [comment]
Good suggestions!

- [x] Adding the subscription binding to Preferences is trivial, I'll do that.
- [ ] I'll take a closer look at the menus and see if I can organize them better.

A bit of context: the hamburger menu is there because the app doesn't have a menu bar on non-Mac operating systems. On the Mac, the native app menu provides access to all (most) of the app features. I've been considering adding a similar conventional menu bar for other desktop OS/environments, which should be even better for discoverability.

One missing little feature is submenus. If I add support for those, the hamburger menu could have a "Page" submenu to match the right-click context menu without crowding the main menu.

**Comment by gawa971** [comment]
Oh I see. I saw the ```#if defined (iPlatformAppleDesktop) ... macos.h ...``` here and there in the code. And the screenshot. But didn't realize how much the appearance changes on Mac OS.

A menu bar would be, like submenus, closer to what we see in other apps and browsers. It's good for discoverability, which is an important goal, considering that gemini is still young. The browser is a factor that will make people stick to gemini. At least I know it did for me, navigating in amfora was painful but it felt pretty good in Lagrange, thanks to the keybindings (opening links with modifier or home-row keys like in qutebrowser is a killer-feature!)

However in my opinion the browser is better without the bar. There are not enough features to justify losing so much space (unless the bar can be hidden). It feels lighter like that way. But I'm biased because I'm currently using a small screen (10.6in, a tablet). By the way I did a test and when the window is super small, a long menu still works: it's possible to scroll in both menus, no graphics bug on that front :+1: 

Of course those are my opinions. You're not responsible for the success or gemini or anything, please just implement what you can and want, the easier that works for you! :)

**Comment by skyjake** [comment]
Version 1.12 adds the option to have a traditional menu bar. This makes all the actions more discoverable. Submenus are still not implemented, though.
